packages/edit-post/src/components/header/header-toolbar/index.native.js

```diff
@@ -83,6 +90,53 @@ function HeaderToolbar( {
 		return isRTL ? buttons.reverse() : buttons;
 	};
 
+	function onInsertBlock( blockType ) {
```

Comment:
If we avoid the inline functions where this is called we can use `useCallback` here. Regarding the internal question about opening the media picker when the block is added, you can check the inserter menu code to use as an example for that functionality.